app_dart/lib/src/datastore/cocoon_config.dart

```diff
@@ -67,20 +75,32 @@ class Config {
     final Uint8List cacheValue = await _cache.getOrCreate(
       configCacheName,
       'flutterBranches',
-      createFn: () => getBranches(Providers.freshHttpClient, loggingService, twoSecondLinearBackoff),
+      createFn: () => getBranches(httpClient, loggingService, twoSecondLinearBackoff),
       ttl: configCacheTtl,
     );
 
     return String.fromCharCodes(cacheValue).split(',');
   }
 
-  // Returns LUCI builders.
-  Future<List<LuciBuilder>> luciBuilders(String bucket, String repo,
-      {String commitSha = 'master', int prNumber}) async {
+  // Returns list of LUCI builders supported in Cocoon.
+  Future<List<LuciBuilder>> luciBuilders(String bucket, String repo, {String commitSha, int prNumber}) async {
     final GithubService githubService = await createGithubService('flutter', repo);
-    return await getLuciBuilders(githubService, Providers.freshHttpClient, twoSecondLinearBackoff, loggingService,
-        RepositorySlug('flutter', repo), bucket,
-        prNumber: prNumber, commitSha: commitSha);
+    final List<LuciBuilder> builders = <LuciBuilder>[];
+    // Grab all builders used by all supported branches.
+    for (String branch in await flutterBranches) {
```

Comment:
Does `try` bucket need to run this loop?

Reply:
FWIW, no longer relevant